models/meshmodel/core/v1alpha1/component.go

```diff
@@ -9,20 +9,30 @@ type TypeMeta struct {
 
 // use NewComponent function for instantiating
 type Component struct {
-	TypeMeta
-	ComponentSpec
-	Metadata map[string]interface{} `json:"metadata,omitempty"`
+	TypeMeta      `gorm:"embedded"`
+	ComponentSpec `gorm:"embedded"`
+	Metadata      map[string]interface{} `json:"metadata,omitempty" gorm:"type:JSONB"`
 	// for backward compatibility
 	Spec string `json:"spec,omitempty"`
 }
-
+type capability struct {
```

Comment:
Please verify that these structs match the Meshery object model table designs.

Reply:
This struct was present in meshery/meshery and is being moved here for a central source of truth.